queries/haskell/highlights.scm

```diff
@@ -11,6 +23,31 @@
 
 (comment) @comment
 
+; FIXME: The below documentation comment queries are inefficient
+; and need to be anchored, using something like
+; ((comment) @_first . (comment)+ @comment.documentation)
+; once https://github.com/neovim/neovim/pull/24738 has been merged.
+;
+; ((comment) @comment.documentation 
+;   (#lua-match? @comment.documentation "^-- |"))
+;
+; ((comment) @_first @comment.documentation 
+;  (comment) @comment.documentation 
+;   (#lua-match? @_first "^-- |"))
+;
+; ((comment) @comment.documentation 
+;   (#lua-match? @comment.documentation "^-- %^"))
+;
+; ((comment) @_first @comment.documentation 
+;  (comment) @comment.documentation 
+;   (#lua-match? @_first "^-- %^"))
+;
+; ((comment) @comment.documentation 
+;   (#lua-match? @comment.documentation "^{-"))
```

Comment:
So is this query. This is also fine

All 
```
((comment) ...
 (comment) @comment.documentation)
```
however, are not, they need anchors and quantifiers